src/main/java/org/sonar/plugins/buildbreaker/QualityGateBreaker.java

```diff
@@ -0,0 +1,195 @@
+import org.apache.commons.lang3.StringUtils;
+import org.sonar.api.CoreProperties;
+import org.sonar.api.batch.BuildBreaker;
+import org.sonar.api.batch.SensorContext;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.config.Settings;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.measures.Metric;
+import org.sonar.api.resources.Project;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarqube.ws.MediaTypes;
+import org.sonarqube.ws.WsCe.TaskResponse;
+import org.sonarqube.ws.WsCe.TaskStatus;
+import org.sonarqube.ws.WsQualityGates.ProjectStatusWsResponse;
+import org.sonarqube.ws.WsQualityGates.ProjectStatusWsResponse.Comparator;
+import org.sonarqube.ws.WsQualityGates.ProjectStatusWsResponse.Condition;
+import org.sonarqube.ws.WsQualityGates.ProjectStatusWsResponse.ProjectStatus;
+import org.sonarqube.ws.WsQualityGates.ProjectStatusWsResponse.Status;
+import org.sonarqube.ws.client.GetRequest;
+import org.sonarqube.ws.client.HttpConnector;
+import org.sonarqube.ws.client.HttpWsClient;
+import org.sonarqube.ws.client.WsClient;
+import org.sonarqube.ws.client.WsRequest;
+import org.sonarqube.ws.client.WsResponse;
+import org.sonarqube.ws.client.qualitygate.ProjectStatusWsRequest;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+public class QualityGateBreaker extends BuildBreaker {
+    private static final String CLASSNAME = QualityGateBreaker.class.getSimpleName();
+    private static final Logger LOGGER = Loggers.get(QualityGateBreaker.class);
+
+    private final FileSystem fileSystem;
+    private final Settings settings;
+
+    public QualityGateBreaker(FileSystem fileSystem, Settings settings) {
+        this.fileSystem = fileSystem;
+        this.settings = settings;
+    }
+
+    @Override
+    public void executeOn(Project project, SensorContext context) {
+        if (!context.analysisMode().isPublish()) {
+            LOGGER.debug("{} is disabled ({} != {})", CLASSNAME, CoreProperties.ANALYSIS_MODE, CoreProperties.ANALYSIS_MODE_PUBLISH);
+            return;
+        }
+
+        if (settings.getBoolean(BuildBreakerPlugin.SKIP_KEY)) {
+            LOGGER.debug("{} is disabled ({} = true)", CLASSNAME, BuildBreakerPlugin.SKIP_KEY);
+            return;
+        }
+
+        checkQualityGate();
+    }
+
+    private void checkQualityGate() {
+        Properties reportTaskProps = loadReportTaskProps();
+
+        HttpConnector httpConnector = new HttpConnector.Builder()
+                .url(reportTaskProps.getProperty("serverUrl"))
+                .credentials(settings.getString(CoreProperties.LOGIN), settings.getString(CoreProperties.PASSWORD))
+                .build();
+
+        String ceTaskId = reportTaskProps.getProperty("ceTaskId");
+        WsRequest ceTaskRequest = new GetRequest("api/ce/task").setParam("id", ceTaskId).setMediaType(MediaTypes.PROTOBUF);
```

Comment:
Will ```MediaTypes.PROTOBUF``` cause compatibility problems with newer versions of SQ?  Should it be switched to ```MediaTypes.JSON```?